From 5ee4e4986c02c67277e224e61f164ce63b0c597c Mon Sep 17 00:00:00 2001
From: Xuefer <xuefer@gmail.com>
Date: Tue, 29 Dec 2015 00:43:16 +0800
Subject: [PATCH] audiopolicy: Handle legacy startOutput on output command
 thread too

* Fixes the build for legacy audio policy
* Broken by: de3b1c01146d2103f6bf56e2659a9c650734945a

Change-Id: I990ce609b3da81c462b35c7dd3f7b11d189e2d2e
Signed-off-by: Xuefer <xuefer@gmail.com>
(cherry picked from commit c20f75882a9af6bd5c7e94c24647453a13cd1d41)
---
 .../service/AudioPolicyInterfaceImplLegacy.cpp           | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/services/audiopolicy/service/AudioPolicyInterfaceImplLegacy.cpp b/services/audiopolicy/service/AudioPolicyInterfaceImplLegacy.cpp
index 575eda0c4..6674f1a33 100644
--- a/services/audiopolicy/service/AudioPolicyInterfaceImplLegacy.cpp
+++ b/services/audiopolicy/service/AudioPolicyInterfaceImplLegacy.cpp
@@ -158,13 +158,27 @@ status_t AudioPolicyService::startOutput(audio_io_handle_t output,
         return NO_INIT;
     }
     ALOGV("startOutput()");
-    // create audio processors according to stream
+    return mOutputCommandThread->startOutputCommand(output, stream, session);
+}
+
+status_t AudioPolicyService::doStartOutput(audio_io_handle_t output,
+                                           audio_stream_type_t stream,
+                                           audio_session_t session)
+{
+    if (uint32_t(stream) >= AUDIO_STREAM_CNT) {
+        return BAD_VALUE;
+    }
+    if (mpAudioPolicy == NULL) {
+        return NO_INIT;
+    }
+    ALOGV("doStartOutput()");
     sp<AudioPolicyEffects>audioPolicyEffects;
     {
         Mutex::Autolock _l(mLock);
         audioPolicyEffects = mAudioPolicyEffects;
     }
     if (audioPolicyEffects != 0) {
+        // create audio processors according to stream
         status_t status = audioPolicyEffects->addOutputSessionEffects(output, stream, session);
         if (status != NO_ERROR && status != ALREADY_EXISTS) {
             ALOGW("Failed to add effects on session %d", session);

